File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02660/p02660_num5_parsed.py
import math

def is_prime(n):
    if n <= 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def max_operations(N):
    prime_powers = set()
    operations = 0
    
    for prime in range(2, int(math.sqrt(N)) + 1):
        if is_prime(prime):
            power = prime
            while power <= N:
                if N % power == 0 and power not in prime_powers:
                    prime_powers.add(power)
                    N /= power
                    operations += 1
                else:
                    break
                
    return operations

N = int(input())
print(max_operations(N))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02660/p02660_num5_parsed.cpp
#include <iostream>

using namespace std;

// Function to check if a number is prime
bool isPrime(int n) {
    if (n <= 1) {
        return false;
    }
    
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    
    return true;
}

// Function to find the maximum number of times the operation can be applied
int findMaxOperations(int N) {
    int operations = 0;
    
    while (N > 1) {
        int maxZ = 1;
        
        // Find the largest prime power divisor of N
        for (int i = 2; i * i <= N; i++) {
            if (isPrime(i) && (N % i == 0)) {
                maxZ = max(maxZ, i);
            }
        }
        
        if (maxZ == 1) {
            break;
        }
        
        // Reduce N by dividing it by maxZ
        N /= maxZ;
        operations++;
    }
    
    return operations;
}

int main() {
    int N;
    cin >> N;
    
    int maxOperations = findMaxOperations(N);
    
    cout << maxOperations << endl;
    
    return 0;
}